src/main/java/htsjdk/samtools/SAMSequenceRecord.java

```diff
@@ -85,8 +87,8 @@ public SAMSequenceRecord(final String name) {
 
     public SAMSequenceRecord(final String name, final int sequenceLength) {
         if (name != null) {
-            if (SEQUENCE_NAME_SPLITTER.matcher(name).find()) {
-                throw new SAMException("Sequence name contains invalid character: " + name);
+            if (!LEGAL_RNAME_PATTERN.matcher(name).useAnchoringBounds(true).matches()) {
```

Comment:
move this into the validate method

Reply:
`SEQUENCE_NAME_SPLITTER` in both code and comments can be removed correct?  I still see it elsewhere.